compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -37,23 +228,157 @@ package #{javaPackage};
+
+            @Override
+            public final int getGenericTypeParameterCount() {
+                return #{length declParams};
+            }
+
+            @Override
+            protected final StructBondType<#{declName}> buildNewInstance(BondType<?>[] genericTypeArguments) {
+                #{buildNewInstanceMethodBody}
+            }
+
+            static void register() {
+                registerStructType(#{declName}.class, new __StructBondTypeBuilderImpl());
+            }
+        }
+
+        #{doubleLineSep 2 fieldDescriptorFieldDecl structFields}
+
+        #{structBondTypeImplConstructorDecl} {
+            #{structBondTypeImplConstructorBody}
+        }
+
+        @Override
+        protected final void initialize() {
+            #{newlineSep 3 genericTypeArgumentVariableDecl declParamsWithIndex}
+            #{newlineSep 3 fieldDescriptorFieldInit structFields}
+            super.initializeBaseAndFields(#{initializeBaseAndFieldsBaseParam}#{initializeBaseAndFieldsFieldsParam});
+        }
+
+        @Override
+        public final java.lang.Class<#{declName}#{params}> getValueClass() {
+            return (java.lang.Class<#{declName}#{params}>) (java.lang.Class<?>) #{declName}.class;
+        }
+
+        @Override
+        public final #{declName}#{params} newInstance() {
+            #{newInstanceMethodBody}
+        }
+
+        @Override
+        protected final void serializeStructFields(com.microsoft.bond.BondType.SerializationContext context, #{declName}#{params} value) throws java.io.IOException {
+            #{newlineSep 3 serializeStructField structFields}
+        }
+
+        @Override
+        protected final void deserializeStructFields(com.microsoft.bond.BondType.TaggedDeserializationContext context, #{declName}#{params} value) throws java.io.IOException {
+            #{newlineSep 3 deserializeStructFieldDeclareBooleans structFields}
+            while (readField(context)) {
+                switch (context.readFieldResult.id) {
+                    #{newlineSep 5 deserializeStructFieldProcessFieldCases structFields}
+                }
+            }
+            #{newlineSep 3 deserializeStructFieldVerifyFields structFields}
+        }
+
+        @Override
+        public final void initializeStructFields(#{declName}#{params} value) {
+            #{newlineSep 3 initializeStructField structFields}
+        }
+    }
+
+    #{bondTypeStaticVariableDecl}
+
+    public static void initializeBondType() {
+        __StructBondTypeImpl.__StructBondTypeBuilderImpl.register();
+    }
+
+    static {
+        initializeBondType();
     }
+    #{bondTypeDescriptorInstanceVariableDecl}
 
-    #{doubleLineSep 1 publicField structFields}
+    #{doubleLineSep 1 publicFieldDecl structFields}
+
+    #{publicConstructorDecl}
+
+    @Override
+    public com.microsoft.bond.StructBondType<? extends com.microsoft.bond.BondSerializable> getBondType() {
+        return #{getBondTypeReturnValue};
+    }
 }|]
             where
```

Comment:
Is discussed, I'll try to split this by Java method.

---
In reply to: [125735980](https://github.com/Microsoft/bond/pull/517#discussion_r125735980) [](ancestors = 125735980,125346136)